### PR TITLE
Breakout target-util into separate thresholds

### PR DESCRIPTION
**Pull request by brycahta**
*Issue #, if available:*
* trying to improve cx

*Description of changes:*
* `--target-utilization` is now `--cpu-threshold` and `--mem-threshold`
* added additional logging for UserConfig and TestFixture
* **witness the widthness**: attempted to improve table readability at the cost of width..

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -226,9 +231,11 @@ The CLI is interrupted after tests began executing on instances, then resumed by
 ### Table Headers
 
 * `INSTANCE TYPE`: instance type
-* `MEETS TARGET UTILIZATION?`: SUCCESS if max CPU and max MEM are less than target utilization; FAIL otherwise
+* `STATUS`: SUCCESS if max CPU and max MEM are less than their respective thresholds; FAIL otherwise
 * `CPU_USAGE_ACTIVE`: max `cpu_usage_active` recorded (p100) over the duration of instance-qualifier run
+* `THRESHOLD`: cpu threshold set by user
```
Comment (haugenj) [inline README.md]: maybe make this one 'CPU_THRESHOLD' and the other one 'MEM_THRESHOLD' ?

File: pkg/config/config.go [hunk 2]
```diff
@@ -168,7 +173,7 @@ in a user friendly format`, binName, binName)
 	if userConfig.Timeout <= 0 {
 		return userConfig, errors.New("you must provide a timeout greater than 0")
 	}
-	fmt.Fprintf(outputStream, "UserConfig: %v\n", userConfig)
+	log.Printf("Starting Instance-Qualifier with User Config: %s\n", userConfig.String())
```
Comment (haugenj) [inline pkg/config/config.go:176]: Does this write it to a log file instead of the cli now?

File: pkg/template/template.go [hunk 3]
```diff
@@ -46,7 +46,7 @@ var (
 
 // UserScript encapsulates the data required for creating user script that will be deployed to the instance(s)
 type UserScript struct {
-	InstanceType, VCpus, Memory, Os, Architecture, BucketName, Timeout, BucketRootDir, TargetUtil, CompressedTestSuiteName, TestSuiteName, CustomScript, Region string
+	InstanceType, VCpus, Memory, Os, Architecture, BucketName, Timeout, BucketRootDir, CompressedTestSuiteName, TestSuiteName, CustomScript, Region string
```
Comment (haugenj) [inline pkg/template/template.go:49]: does this not need the new threshold values in place of TargetUtil?

**Review by haugenj — APPROVED** [review]
Looks good to me, just put down a couple questions to clarify my understanding

File: pkg/config/config.go (hunk 2 above)
Comment (brycahta) [reply pkg/config/config.go:176]: No, still prints in the terminal (there's currently no log file for IQ itself):

```

➜  amazon-ec2-instance-qualifier git:(improve-cx) ✗ ./build/ec2-instance-qualifier --config-file=/Users/crtbry/workplace/go/src/iq_brycahta_fork_ws/amazon-ec2-instance-qualifier/2_instances.json
2020/10/09 10:46:28 Starting Instance-Qualifier with User Config: 
                InstanceTypes: m4.large,
                TestSuiteName: /Users/crtbry/workplace/go/src/iq_brycahta_fork_ws/amazon-ec2-instance-qualifier/test/app-for-e2e/tests,
                CpuThreshold: 80,
                MemThreshold: 10,
                VpcId: ,
                SubnetId: ,
                AmiId: ,
                Timeout: 600,
                Persist: false,
                Profile: ,
                Region: us-east-2,
                Bucket: ,
                CustomScriptPath: /Users/crtbry/workplace/go/src/iq_brycahta_fork_ws/amazon-ec2-instance-qualifier/test/templates/custom-script_sample.template,
                ConfigFilePath: /Users/crtbry/workplace/go/src/iq_brycahta_fork_ws/amazon-ec2-instance-qualifier/2_instances.json

```

File: pkg/template/template.go (hunk 3 above)
Comment (brycahta) [reply pkg/template/template.go:49]: Correct it does not. Before transitioning to using CloudWatch metrics, the Agent code (run on the tested instance) would be responsible for aggregating the data AND determining whether or not it met the TargetUtil.

Now that IQ queries CloudWatch outside of Agent code, the comparison to thresholds will be handled locally (i.e. not on the tested instance).

File: README.md (hunk 1 above)
Comment (brycahta) [reply README.md]: Yeah, I was trying to make the table narrower but definitely not worth at the cost of clarity. will address in revision

**Review by bwagner5 — APPROVED** [review]
LGTM, nice work!

**Pull request merged**